101-vm-simple-linux/azuredeploy.json

```diff
@@ -144,7 +166,8 @@
         "osProfile": {
           "computerName": "[variables('vmName')]",
           "adminUsername": "[parameters('adminUsername')]",
-          "adminPassword": "[parameters('adminPassword')]"
+          "adminPassword": "[if(equals(parameters('authenticationType'), 'password'), parameters('adminPasswordOrKey'), json('null'))]",
```

Comment:
Simplify this and just use:

"adminPassword": "[parameters('adminPasswordOrKey')]",

Since disablePasswordAuth is true, it will be ignored.